### PR TITLE
ShopUser class is now EncoderAware to provide more flexibility for ch…

### PR DESCRIPTION
**Pull request by kortwotze**
…oosing how to encode passwords

| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | mentioned in #9560 
| License         | MIT

### CODE REVIEW

**Comment by kortwotze** [comment]
Can someone help me finding out which test-configuration to change to add the new encoder from security.yml to the test-environments?

**Comment by kortwotze** [comment]
CC @pamil, because this references his issue.

**Comment by pamil** [comment]
Rebased the PR so that the build should be passing now.

File: config/packages/security.yaml [hunk 1]
```diff
@@ -10,7 +10,8 @@ security:
         sylius_shop_user_provider:
             id: sylius.shop_user_provider.email_or_name_based
     encoders:
-        Sylius\Component\User\Model\UserInterface: sha512
+        sylius_default_password_encoder:
```
Comment (pamil) [inline config/packages/security.yaml]: I would name it `sha512` (`default` looks tricky and might break the application if someone suddenly changes the algorithm).

File: config/packages/security.yaml [hunk 2]
```diff
@@ -10,7 +10,8 @@ security:
...
+            id: 'sylius.security.user_password_encoder'
```
Comment (pamil) [inline config/packages/security.yaml]: Same here - `sylius.security.user_password_encoder.sha512`?

File: config/packages/security.yaml (hunk 1 above)
Comment (kortwotze) [reply config/packages/security.yaml]: Actually, I would leave the encoder name as is and change the service ID of the "default" encoder. So the default encoder is configured to be the sha512 one

File: src/Sylius/Bundle/UserBundle/Tests/Functional/app/config/config.yml [hunk 3]
```diff
@@ -61,8 +61,6 @@ security:
     encoders:
         sylius_password_encoder_sha512:
             id: 'sylius.security.user_password_encoder.sha512'
-        sylius_password_encoder_argon2i:
```
Comment (kortwotze) [inline src/Sylius/Bundle/UserBundle/Tests/Functional/app/config/config.yml]: This is intended, after adding and committing I tried to check if argon2i is actually enabled in my sylius container. Seeing that it was not, I chose not to take any risk and remove it again.

It most likely will come back sooner or later. :)

**Comment by pamil** [comment]
Thanks, Peter! :1st_place_medal:

**Pull request merged**